clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -439,6 +439,9 @@ def warn_deprecated_literal_operator_id: Warning<
   "is deprecated">, InGroup<DeprecatedLiteralOperator>, DefaultIgnore;
 def warn_reserved_module_name : Warning<
   "%0 is a reserved name for a module">, InGroup<ReservedModuleIdentifier>;
+def warn_import_implementation_partition_unit_in_interface_unit : Warning<
+  "it is not suggested to import implementation partition unit in interface unit">,
```

Comment:
```suggestion
  "importing an implementation partition unit in a module interface is not recommended. Symbols exported by %0 may not be reachable">,
```
Maybe something like that ?

Reply:
Done. But I changed `Symbols exported ` to `Names in ` since the term `name` is more precise and we won't export things in implementation partition unit.